src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1101 @@
+};
+
+class EntryIteratorDelegate : public EntryIterator::Delegate
+{
+public:
+    // Pool support
+    static EntryIteratorDelegate pool[kEntryIteratorDelegatePoolSize];
+
+    static EntryIteratorDelegate * Find(EntryIterator::Delegate & candidate)
+    {
+        if (InPool(candidate))
+        {
+            return static_cast<EntryIteratorDelegate *>(&candidate);
+        }
+        for (auto & delegate : pool)
+        {
+            if (!delegate.InUse())
+            {
+                return &delegate;
+            }
+        }
+        return nullptr;
+    }
+
+    static bool InPool(const EntryIterator::Delegate & delegate)
+    {
+        constexpr auto end = pool + ArraySize(pool);
+        return pool <= &delegate && &delegate < end;
+    }
+
+public:
+    void Release() override { mInUse = false; }
+
+    CHIP_ERROR Next(Entry & entry) override
+    {
+        constexpr auto & acl = EntryStorage::acl;
+        constexpr auto end = acl + ArraySize(acl);
+        while (true)
+        {
+            if (mStorage == nullptr)
+            {
+                mStorage = acl;
+            }
+            else if (mStorage < end)
+            {
+                mStorage++;
+            }
+            if (mStorage == end || !mStorage->InUse())
+            {
+                mStorage = end;
+                break;
+            }
+            if (mFabricFiltered && mStorage->mFabricIndex != mFabricIndex)
+            {
+                continue;
+            }
+            if (auto delegate = EntryDelegate::Find(entry.GetDelegate()))
+            {
+                delegate->Init(entry, *mStorage);
+                return CHIP_NO_ERROR;
+            }
+            return CHIP_ERROR_BUFFER_TOO_SMALL;
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+public:
+    void Init(EntryIterator & iterator, const FabricIndex * fabricIndex)
+    {
+        iterator.SetDelegate(*this);
+        mInUse = true;
+        mFabricFiltered = fabricIndex != nullptr;
+        if (mFabricFiltered)
+        {
+            mFabricIndex = *fabricIndex;
+        }
+        mStorage = nullptr;
+    }
+
+    bool InUse() const { return mInUse; }
+
+    void FixAfterDelete(EntryStorage & storage)
+    {
+        constexpr auto & acl = EntryStorage::acl;
+        constexpr auto end = acl + ArraySize(acl);
+        if (&storage <= mStorage && mStorage < end)
+        {
+            if (mStorage == acl)
+            {
+                mStorage = nullptr;
+            }
+            else
+            {
+                mStorage--;
+            }
+        }
+    }
+
+private:
+    bool mInUse = false;
+    bool mFabricFiltered;
+    FabricIndex mFabricIndex;
+    EntryStorage * mStorage;
+};
+
+CHIP_ERROR CopyViaInterface(const Entry & entry, EntryStorage & storage)
+{
+#if CHIP_CONFIG_EXAMPLE_ACCESS_CONTROL_FOREIGN_DELEGATE_SUPPORT
+    // NOTE: function uses a fair amount of stack
+    //       and is only necessary if using foreign delegates
```

Comment:
What is a "foreign delegate"

Reply:
Any Entry::Delegate that isn't an example EntryDelegate (as in this file). So far none exist, you won't have one unless you write one and start using it. Even then, you only need this function if you are using this example code. I.e. if don't use the example code at all, the point is moot, you're free to do that. This code is only here for completeness, it's not needed if just using the example AccessControl::Delegate.

Reply:
I made this clearer in the comment.